pkg/cri/server/base: log "id" and "type" for CRI plugin startup

thaJeztah opened the pull request:
- follow-up to https://github.com/containerd/containerd/pull/10145


### pkg/cri/server/base: log "id" and "type" for CRI plugin startup

Log the id, and type as a field as part of the log message.

The CRI plugin was moved from an external plugin to an internal plugin,
but for backward compatibility, the plugin is still referred to using
its "legacy" ID for public-facing things, such as the config-file and
state directories.

This patch adds two consts (pluginID and legacyPluginID) to document this.

Before this (config truncated for readability):

    INFO[2023-12-07T15:27:15.862946138Z] loading plugin                                id=io.containerd.internal.v1.tracing type=io.containerd.internal.v1
    INFO[2023-12-07T15:27:15.862946138Z] loading plugin                                id=io.containerd.internal.v1.cri type=io.containerd.internal.v1
    INFO[2023-12-07T15:27:15.862946138Z] starting cri plugin                           config="{\"...\":{}}"

After this:

    INFO[2023-12-07T15:27:15.862946138Z] loading plugin                                id=io.containerd.internal.v1.tracing type=io.containerd.internal.v1
    INFO[2023-12-07T15:27:15.862946138Z] loading plugin                                id=io.containerd.internal.v1.cri type=io.containerd.internal.v1
    INFO[2023-12-07T15:27:15.862946138Z] starting cri plugin                           config="{\"...\":{}}" id=io.containerd.internal.v1.cri type=io.containerd.internal.v1

thaJeztah commented:
I put the JSON change in a separate commit; open for discussion on that one, but I was looking at the logs, and thought that it may be more useful for users to get the JSON presentation for this (at the cost of harder to read due to the escaping).

/cc @mikebrow

dmcgowan commented on pkg/cri/server/base/cri_base.go:
```diff
@@ -106,7 +108,9 @@ func initCRIBase(ic *plugin.InitContext) (interface{}, error) {
-	log.G(ctx).Infof("Start cri plugin with config %+v", c)
+	// Ignoring errors here; this should never fail.
+	cfg, _ := json.Marshal(c)
+	log.G(ctx).WithFields(log.Fields{"config": string(cfg), "id": pluginID, "type": plugins.InternalPlugin}).Info("starting cri plugin")
```
I like using structured, but the id and plugin type actually don't match here. The `pluginID` is using the old id for compatibility. The relevant plugin is `io.containerd.internal.cri`, but I'm not so sure it is useful to show that.

fuweid commented on pkg/cri/server/base/cri_base.go:
```diff
@@ -47,6 +47,8 @@ type CRIBase struct {
 	BaseOCISpecs map[string]*oci.Spec
 }
 
+const pluginID = "io.containerd.grpc.v1.cri"
```
I think the `pluginID` should `requiredPluginID`.

thaJeztah commented:
Updated 👍 

- removed the const
- removed the `id` and `type` fields from the logs

(will open a follow-up PR to continue the discussion from above)

thaJeztah commented:
And I just realised that adding `id` and `type` on its own may not be that controversial (I noticed the `plugin load` already has those); 

```
INFO[2023-12-07T15:33:39.914580427Z] loading plugin                                id=io.containerd.internal.v1.cri type=io.containerd.internal.v1
```

I guess I might as well keep that commit as part of this PR (but open a ticket to discuss if we want to migrate)

thaJeztah commented:
updated; PTAL (@dmcgowan let me know if you still have concerns about including `id` and `type`; see above for my reason to include it (as it's included in the `load plugin` logs as well))

thaJeztah commented:
Looks like the last commit still needed further discussion; I moved the first 2 commits to a separate PR;

- https://github.com/containerd/containerd/pull/10145
